src/librustc/hir/check_attr.rs

```diff
@@ -428,21 +431,29 @@ impl CheckAttrVisitor<'tcx> {
         // Error on repr(transparent, <anything else>).
         if is_transparent && hints.len() > 1 {
             let hint_spans: Vec<_> = hint_spans.clone().collect();
-            span_err!(
+            struct_span_err!(
                 self.tcx.sess,
                 hint_spans,
                 E0692,
                 "transparent {} cannot have other repr hints",
                 target
-            );
+            )
+            .emit();
         }
         // Warn on repr(u8, u16), repr(C, simd), and c-like-enum-repr(C, u8)
         if (int_reprs > 1)
             || (is_simd && is_c)
             || (int_reprs == 1 && is_c && item.map_or(false, |item| is_c_like_enum(item)))
         {
-            let hint_spans: Vec<_> = hint_spans.collect();
-            span_warn!(self.tcx.sess, hint_spans, E0566, "conflicting representation hints");
```

Comment:
I suspect this was made a warning rather than an error for backward compatibility.
So it's rather a deprecation lint than a linty lint (or perhaps it can be made an error right away).

Reply:
The warning was introduced in https://github.com/rust-lang/rust/pull/34623. I got the sense that it was deliberate as "not going to become an error" but I could be wrong.

Reply:
https://github.com/rust-lang/rust/pull/34623 started as an error (https://github.com/rust-lang/rust/pull/34623#issuecomment-230188553) and things like `repr(u8, u16)` can never work as stated, so it never makes sense to `allow` this lint long term, so it looks like a deprecation lint to me.

Reply:
Ah, that makes sense. I'll go make it an error and we can find out whether 4 years later, anyone still depends on it in the beta crater runs...?

Reply:
Fixed.